packages/core/src/pages/index.tsx

```diff
@@ -100,31 +100,43 @@ export const getStaticProps: GetStaticProps<
   Record<string, string>,
   Locator
 > = async ({ previewData }) => {
-  const serverData = await getDynamicContent({ pageType: 'home' })
-  const globalSections = await getGlobalSectionsData(previewData)
+  const globalSectionsPromise = getGlobalSectionsData(previewData)
+  const serverDataPromise = getDynamicContent({ pageType: 'home' })
 
   if (storeConfig.cms.data) {
     const cmsData = JSON.parse(storeConfig.cms.data)
     const page = cmsData['home'][0]
 
     if (page) {
-      const pageData = await getPage<PageContentType>({
+      const pageDataPromise = getPage<PageContentType>({
         contentType: 'home',
         documentId: page.documentId,
         versionId: page.versionId,
       })
 
+      const [pageData, globalSections, serverData] = await Promise.all([
+        pageDataPromise,
+        globalSectionsPromise,
+        serverDataPromise,
+      ])
+
       return {
         props: { page: pageData, globalSections, serverData },
       }
     }
   }
 
-  const page = await getPage<PageContentType>({
+  const pagePromise = getPage<PageContentType>({
     ...(previewData?.contentType === 'home' && previewData),
     contentType: 'home',
   })
 
+  const [page, globalSections, serverData] = await Promise.all([
+    pagePromise,
+    globalSectionsPromise,
+    serverDataPromise,
+  ])
+
   return {
     props: { page, globalSections, serverData },
   }
```

Comment:
suggestion (non-blocking): since the code inside and outside the `if` is so similar, what do you think about reducing the code inside the `if`?

```suggestion
  let cmsPage = null

  if (storeConfig.cms.data) {
    const cmsData = JSON.parse(storeConfig.cms.data)
    cmsPage = cmsData['home'][0]
  }

  const pagePromise = cmsPage
    ? getPage<PageContentType>({
        contentType: 'home',
        documentId: cmsPage.documentId,
        versionId: cmsPage.versionId,
      })
    : getPage<PageContentType>({
        ...(previewData?.contentType === 'home' && previewData),
        contentType: 'home',
      })

  const [page, globalSections, serverData] = await Promise.all([
    pagePromise,
    globalSectionsPromise,
    serverDataPromise,
  ])

  return {
    props: { page, globalSections, serverData },
  }
```

Reply:
Nice Improvement!